Unit-Test fails if netty dependency is missing

Hey, 

after #734 the Unit-Test fails if netty is missing on the classpath. 

Reproducer: [Branch testJunit](https://github.com/quarkiverse/quarkus-amazon-services/compare/main...knuspertante:quarkus-amazon-services:testJunit?expand=1#diff-1eb679daf6b62d0e1a79c3923b2ecc752cccc253fcc0010704e3bca2c5865195) Maven Module [integration-tests-dynamodb-without-async-client](https://github.com/quarkiverse/quarkus-amazon-services/compare/main...knuspertante:quarkus-amazon-services:testJunit?expand=1#diff-1eb679daf6b62d0e1a79c3923b2ecc752cccc253fcc0010704e3bca2c5865195)

StackTrace
```sh
java.lang.RuntimeException: java.lang.ExceptionInInitializerError

	at io.quarkus.test.junit.QuarkusTestExtension.throwBootFailureException(QuarkusTestExtension.java:640)
	at io.quarkus.test.junit.QuarkusTestExtension.interceptTestClassConstructor(QuarkusTestExtension.java:711)
	at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.lambda$invoke$0(InterceptingExecutableInvoker.java:93)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
	at org.junit.jupiter.api.extension.InvocationInterceptor.interceptTestClassConstructor(InvocationInterceptor.java:73)
	at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.lambda$invoke$0(InterceptingExecutableInvoker.java:93)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.proceed(InvocationInterceptorChain.java:64)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.chainAndInvoke(InvocationInterceptorChain.java:45)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.invoke(InvocationInterceptorChain.java:37)
	at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.invoke(InterceptingExecutableInvoker.java:92)
	at org.junit.jupiter.engine.execution.InterceptingExecutableInvoker.invoke(InterceptingExecutableInvoker.java:62)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.invokeTestClassConstructor(ClassBasedTestDescriptor.java:363)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.instantiateTestClass(ClassBasedTestDescriptor.java:310)
	at org.junit.jupiter.engine.descriptor.ClassTestDescriptor.instantiateTestClass(ClassTestDescriptor.java:79)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.instantiateAndPostProcessTestInstance(ClassBasedTestDescriptor.java:286)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.lambda$testInstancesProvider$4(ClassBasedTestDescriptor.java:278)
	at java.base/java.util.Optional.orElseGet(Optional.java:364)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.lambda$testInstancesProvider$5(ClassBasedTestDescriptor.java:277)
	at org.junit.jupiter.engine.execution.TestInstancesProvider.getTestInstances(TestInstancesProvider.java:31)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$prepare$0(TestMethodTestDescriptor.java:105)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.prepare(TestMethodTestDescriptor.java:104)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.prepare(TestMethodTestDescriptor.java:68)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$prepare$2(NodeTestTask.java:123)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.prepare(NodeTestTask.java:123)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:90)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:35)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:54)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:147)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:127)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:90)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:55)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:102)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:54)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:114)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:86)
	at org.junit.platform.launcher.core.DefaultLauncherSession$DelegatingLauncher.execute(DefaultLauncherSession.java:86)
	at org.junit.platform.launcher.core.SessionPerRequestLauncher.execute(SessionPerRequestLauncher.java:53)
	at com.intellij.junit5.JUnit5IdeaTestRunner.startRunnerWithArgs(JUnit5IdeaTestRunner.java:57)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater$1.execute(IdeaTestRunner.java:38)
	at com.intellij.rt.execution.junit.TestsRepeater.repeat(TestsRepeater.java:11)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:35)
	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:232)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:55)
Caused by: java.lang.ExceptionInInitializerError
	at io.quarkus.runner.ApplicationImpl.<clinit>(Unknown Source)
	at java.base/java.lang.Class.forName0(Native Method)
	at java.base/java.lang.Class.forName(Class.java:495)
	at java.base/java.lang.Class.forName(Class.java:474)
	at io.quarkus.runner.bootstrap.StartupActionImpl.run(StartupActionImpl.java:257)
	at io.quarkus.test.junit.QuarkusTestExtension.doJavaStart(QuarkusTestExtension.java:251)
	at io.quarkus.test.junit.QuarkusTestExtension.ensureStarted(QuarkusTestExtension.java:607)
	at io.quarkus.test.junit.QuarkusTestExtension.beforeAll(QuarkusTestExtension.java:655)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.lambda$invokeBeforeAllCallbacks$12(ClassBasedTestDescriptor.java:395)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.invokeBeforeAllCallbacks(ClassBasedTestDescriptor.java:395)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.before(ClassBasedTestDescriptor.java:211)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.before(ClassBasedTestDescriptor.java:84)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:148)
	... 36 more
	Suppressed: java.lang.NoClassDefFoundError: Could not initialize class io.quarkus.runtime.generated.Config
		at java.base/java.lang.Class.forName0(Native Method)
		at java.base/java.lang.Class.forName(Class.java:495)
		at java.base/java.lang.Class.forName(Class.java:474)
		at io.quarkus.runner.bootstrap.StartupActionImpl.run(StartupActionImpl.java:261)
		... 45 more
	Caused by: java.lang.ExceptionInInitializerError: Exception java.lang.ClassNotFoundException: io.netty.handler.ssl.SslProvider [in thread "main"]
		at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
		at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188)
		at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
		at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:516)
		at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:466)
		at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:516)
		at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:466)
		at java.base/java.lang.Class.forName0(Native Method)
		at java.base/java.lang.Class.forName(Class.java:495)
		at java.base/java.lang.Class.forName(Class.java:474)
		at io.quarkus.runtime.generated.Config.<clinit>(Unknown Source)
		at io.quarkus.runner.ApplicationImpl.<clinit>(Unknown Source)
		at java.base/java.lang.Class.forName0(Native Method)
		at java.base/java.lang.Class.forName(Class.java:495)
		at java.base/java.lang.Class.forName(Class.java:474)
		at io.quarkus.runner.bootstrap.StartupActionImpl.run(StartupActionImpl.java:257)
		... 45 more
Caused by: java.lang.ClassNotFoundException: io.netty.handler.ssl.SslProvider
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:188)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
	at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:516)
	at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:466)
	at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:516)
	at io.quarkus.bootstrap.classloading.QuarkusClassLoader.loadClass(QuarkusClassLoader.java:466)
	at java.base/java.lang.Class.forName0(Native Method)
	at java.base/java.lang.Class.forName(Class.java:495)
	at java.base/java.lang.Class.forName(Class.java:474)
	at io.quarkus.runtime.generated.Config.<clinit>(Unknown Source)
	... 50 more
```

Is this the way it should be? I suggest du add netty if it is necessary? 
Pls feel free to discuss.

Kind regards
Michael

I'm facing similar issue. not in test but on aws-lambda. 
using quarkus-amazon-lambda (as sqs handler) + quarkus-amazon-dynamodb, + few other quarkus-amazon exts.
should I add 'io.netty:netty-handler' dependency as a workaround?  

```java.lang.ExceptionInInitializerError: java.lang.ExceptionInInitializerError
java.lang.ExceptionInInitializerError
	at io.quarkus.runner.ApplicationImpl.<clinit>(Unknown Source)
	at java.base/java.lang.Class.forName0(Native Method)
	at java.base/java.lang.Class.forName(Unknown Source)
	at io.quarkus.runtime.Quarkus.manualInitialize(Quarkus.java:236)
	at io.quarkus.amazon.lambda.runtime.QuarkusStreamHandler.<init>(QuarkusStreamHandler.java:21)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(Unknown Source)
	at java.base/java.lang.reflect.Constructor.newInstance(Unknown Source)
Caused by: java.lang.ClassNotFoundException: io.netty.handler.ssl.SslProvider. Current classpath: file:/var/task/:file:/var/task/lib/com.amazonaws.aws-lambda-java-core-1.2.2.jar:file:/var/task/lib/com.amazonaws.aws-lambda-java-events-3.11.1.jar:file:/var/task/lib/com.fasterxml.jackson.core.jackson-annotations-2.15.0.jar:file:/var/task/lib/com.fasterxml.jackson.core.jackson-core-2.15.0.jar:file:/var/task/lib/com.fasterxml.jackson.datatype.jackson-datatype-jdk8-2.15.0.jar:file:/var/task/lib/com.fasterxml.jackson.datatype.jackson-datatype-joda-2.15.0.jar:file:/var/task/lib/com.fasterxml.jackson.datatype.jackson-datatype-jsr310-2.15.0.jar:file:/var/task/lib/com.fasterxml.jackson.module.jackson-module-parameter-names-2.15.0.jar:file:/var/task/lib/com.google.errorprone.error_prone_annotations-2.19.1.jar:file:/var/task/lib/com.google.guava.failureaccess-1.0.1.jar:file:/var/task/lib/com.google.guava.guava-31.1-jre.jar:file:/var/task/lib/com.google.j2objc.j2objc-annotations-1.3.jar:file:/var/task/lib/commons-collections.commons-collections-3.2.2.jar:file:/var/task/lib/io.github.crac.org-crac-0.1.3.jar:file:/var/task/lib/io.quarkiverse.amazonservices.quarkus-amazon-common-2.3.0.jar:file:/var/task/lib/io.quarkiverse.loggingjson.quarkus-logging-json-2.0.0.jar:file:/var/task/lib/io.quarkus.arc.arc-3.1.0.Final.jar:file:/var/task/lib/io.quarkus.quarkus-amazon-lambda-3.1.0.Final.jar:file:/var/task/lib/io.quarkus.quarkus-amazon-lambda-common-3.1.0.Final.jar:file:/var/task/lib/io.quarkus.quarkus-arc-3.1.0.Final.jar:file:/var/task/lib/io.quarkus.quarkus-bootstrap-runner-3.1.0.Final.jar:file:/var/task/lib/io.quarkus.quarkus-core-3.1.0.Final.jar:file:/var/task/lib/io.quarkus.quarkus-development-mode-spi-3.1.0.Final.jar:file:/var/task/lib/io.quarkus.quarkus-fs-util-0.0.9.jar:file:/var/task/lib/io.quarkus.quarkus-jackson-3.1.0.Final.jar:file:/var/task/lib/io.quarkus.quarkus-jsonb-3.1.0.Final.jar:file:/var/task/lib/io.quarkus.quarkus-jsonp-3.1.0.Final.jar:file:/var/task/lib/io.smallrye.common.smallrye-common-annotation-2.1.0.jar:file:/var/task/lib/io.smallrye.common.smallrye-common-classloader-2.1.0.jar:file:/var/task/lib/io.smallrye.common.smallrye-common-constraint-2.1.0.jar:file:/var/task/lib/io.smallrye.common.smallrye-common-expression-2.1.0.jar:file:/var/task/lib/io.smallrye.common.smallrye-common-function-2.1.0.jar:file:/var/task/lib/io.smallrye.common.smallrye-common-io-2.1.0.jar:file:/var/task/lib/io.smallrye.common.smallrye-common-os-2.1.0.jar:file:/var/task/lib/io.smallrye.config.smallrye-config-3.2.1.jar:file:/var/task/lib/io.smallrye.config.smallrye-config-common-3.2.1.jar:file:/var/task/lib/io.smallrye.config.smallrye-config-core-3.2.1.jar:file:/var/task/lib/io.smallrye.reactive.mutiny-2.1.0.jar:file:/var/task/lib/jakarta.annotation.jakarta.annotation-api-2.1.1.jar:file:/var/task/lib/jakarta.el.jakarta.el-api-5.0.1.jar:file:/var/task/lib/jakarta.enterprise.jakarta.enterprise.cdi-api-4.0.1.jar:file:/var/task/lib/jakarta.enterprise.jakarta.enterprise.lang-model-4.0.1.jar:file:/var/task/lib/jakarta.inject.jakarta.inject-api-2.0.1.jar:file:/var/task/lib/jakarta.interceptor.jakarta.interceptor-api-2.1.0.jar:file:/var/task/lib/jakarta.json.bind.jakarta.json.bind-api-3.0.0.jar:file:/var/task/lib/jakarta.json.jakarta.json-api-2.1.1.jar:file:/var/task/lib/jakarta.transaction.jakarta.transaction-api-2.0.1.jar:file:/var/task/lib/javax.inject.javax.inject-1.jar:file:/var/task/lib/joda-time.joda-time-2.10.14.jar:file:/var/task/lib/modified-com.fasterxml.jackson.core.jackson-databind-2.15.0.jar:file:/var/task/lib/modified-io.quarkiverse.amazonservices.quarkus-amazon-dynamodb-2.3.0.jar:file:/var/task/lib/modified-io.quarkiverse.amazonservices.quarkus-amazon-sqs-2.3.0.jar:file:/var/task/lib/modified-io.quarkiverse.amazonservices.quarkus-amazon-sts-2.3.0.jar:file:/var/task/lib/org.apache.commons.commons-lang3-3.12.0.jar:file:/var/task/lib/org.apache.logging.log4j.log4j-api-2.20.0.jar:file:/var/task/lib/org.checkerframework.checker-qual-3.32.0.jar:file:/var/task/lib/org.eclipse.microprofile.config.microprofile-config-api-3.0.3.jar:file:/var/task/lib/org.eclipse.microprofile.context-propagation.microprofile-context-propagation-api-1.3.jar:file:/var/task/lib/org.eclipse.parsson.parsson-1.1.1.jar:file:/var/task/lib/org.eclipse.yasson-3.0.3.jar:file:/var/task/lib/org.graalvm.sdk.graal-sdk-22.3.2.jar:file:/var/task/lib/org.jboss.logging.commons-logging-jboss-logging-1.0.0.Final.jar:file:/var/task/lib/org.jboss.logging.jboss-logging-3.5.0.Final.jar:file:/var/task/lib/org.jboss.logging.jboss-logging-annotations-2.2.1.Final.jar:file:/var/task/lib/org.jboss.logmanager.jboss-logmanager-embedded-1.1.1.jar:file:/var/task/lib/org.jboss.slf4j.slf4j-jboss-logmanager-2.0.0.Final.jar:file:/var/task/lib/org.jboss.threads.jboss-threads-3.5.0.Final.jar:file:/var/task/lib/org.reactivestreams.reactive-streams-1.0.4.jar:file:/var/task/lib/org.slf4j.slf4j-api-2.0.6.jar:file:/var/task/lib/org.wildfly.common.wildfly-common-1.5.4.Final-format-001.jar:file:/var/task/lib/software.amazon.awssdk.annotations-2.20.72.jar:file:/var/task/lib/software.amazon.awssdk.auth-2.20.72.jar:file:/var/task/lib/software.amazon.awssdk.aws-core-2.20.72.jar:file:/var/task/lib/software.amazon.awssdk.aws-json-protocol-2.20.72.jar:file:/var/task/lib/software.amazon.awssdk.aws-query-protocol-2.20.72.jar:file:/var/task/lib/software.amazon.awssdk.dynamodb-2.20.72.jar:file:/var/task/lib/software.amazon.awssdk.endpoints-spi-2.20.72.jar:file:/var/task/lib/software.amazon.awssdk.http-client-spi-2.20.72.jar:file:/var/task/lib/software.amazon.awssdk.json-utils-2.20.72.jar:file:/var/task/lib/software.amazon.awssdk.metrics-spi-2.20.72.jar:file:/var/task/lib/software.amazon.awssdk.profiles-2.20.72.jar:file:/var/task/lib/software.amazon.awssdk.protocol-core-2.20.72.jar:file:/var/task/lib/software.amazon.awssdk.regions-2.20.72.jar:file:/var/task/lib/software.amazon.awssdk.sdk-core-2.20.72.jar:file:/var/task/lib/software.amazon.awssdk.sqs-2.20.72.jar:file:/var/task/lib/software.amazon.awssdk.sts-2.20.72.jar:file:/var/task/lib/software.amazon.awssdk.third-party-jackson-core-2.20.72.jar:file:/var/task/lib/software.amazon.awssdk.url-connection-client-2.20.72.jar:file:/var/task/lib/software.amazon.awssdk.utils-2.20.72.jar:file:/var/task/lib/software.amazon.eventstream.eventstream-1.0.1.jar

```

--- a/common/runtime/src/main/java/io/quarkus/amazon/common/runtime/AmazonClientNettyTransportRecorder.java
+++ b/common/runtime/src/main/java/io/quarkus/amazon/common/runtime/AmazonClientNettyTransportRecorder.java
@@ -3,6 +3,7 @@ package io.quarkus.amazon.common.runtime;
 import java.util.Collections;
 import java.util.HashSet;
 
+import io.quarkus.amazon.common.runtime.NettyHttpClientConfig.SslProviderType;
 import io.quarkus.runtime.RuntimeValue;
 import io.quarkus.runtime.annotations.Recorder;
 import software.amazon.awssdk.http.TlsTrustManagersProvider;
@@ -33,7 +34,7 @@ public class AmazonClientNettyTransportRecorder extends AbstractAmazonClientTran
         builder.readTimeout(asyncConfig.readTimeout);
         builder.writeTimeout(asyncConfig.writeTimeout);
         builder.tcpKeepAlive(asyncConfig.tcpKeepAlive);
-        asyncConfig.sslProvider.ifPresent(builder::sslProvider);
+        asyncConfig.sslProvider.map(SslProviderType::create).ifPresent(builder::sslProvider);
         builder.useIdleConnectionReaper(asyncConfig.useIdleConnectionReaper);
 
         if (asyncConfig.http2.initialWindowSize.isPresent() || asyncConfig.http2.maxStreams.isPresent()) {

--- a/common/runtime/src/main/java/io/quarkus/amazon/common/runtime/NettyHttpClientConfig.java
+++ b/common/runtime/src/main/java/io/quarkus/amazon/common/runtime/NettyHttpClientConfig.java
@@ -17,7 +17,8 @@ public class NettyHttpClientConfig {
     /**
      * The maximum number of allowed concurrent requests.
      * <p>
-     * For HTTP/1.1 this is the same as max connections. For HTTP/2 the number of connections that will be used depends on the
+     * For HTTP/1.1 this is the same as max connections. For HTTP/2 the number of
+     * connections that will be used depends on the
      * max streams allowed per connection.
      */
     @ConfigItem(defaultValue = "50")
@@ -32,7 +33,8 @@ public class NettyHttpClientConfig {
     public int maxPendingConnectionAcquires;
 
     /**
-     * The amount of time to wait for a read on a socket before an exception is thrown.
+     * The amount of time to wait for a read on a socket before an exception is
+     * thrown.
      * <p>
      * Specify `0` to disable.
      */
@@ -40,7 +42,8 @@ public class NettyHttpClientConfig {
     public Duration readTimeout;
 
     /**
-     * The amount of time to wait for a write on a socket before an exception is thrown.
+     * The amount of time to wait for a write on a socket before an exception is
+     * thrown.
      * <p>
      * Specify `0` to disable.
      */
@@ -48,27 +51,32 @@ public class NettyHttpClientConfig {
     public Duration writeTimeout;
 
     /**
-     * The amount of time to wait when initially establishing a connection before giving up and timing out.
+     * The amount of time to wait when initially establishing a connection before
+     * giving up and timing out.
      */
     @ConfigItem(defaultValue = "10S")
     public Duration connectionTimeout;
 
     /**
-     * The amount of time to wait when acquiring a connection from the pool before giving up and timing out.
+     * The amount of time to wait when acquiring a connection from the pool before
+     * giving up and timing out.
      */
     @ConfigItem(defaultValue = "2S")
     public Duration connectionAcquisitionTimeout;
 
     /**
-     * The maximum amount of time that a connection should be allowed to remain open, regardless of usage frequency.
+     * The maximum amount of time that a connection should be allowed to remain
+     * open, regardless of usage frequency.
      */
     @ConfigItem
     public Optional<Duration> connectionTimeToLive;
 
     /**
-     * The maximum amount of time that a connection should be allowed to remain open while idle.
+     * The maximum amount of time that a connection should be allowed to remain open
+     * while idle.
      * <p>
-     * Currently has no effect if `quarkus.<amazon-service>.async-client.use-idle-connection-reaper` is false.
+     * Currently has no effect if
+     * `quarkus.<amazon-service>.async-client.use-idle-connection-reaper` is false.
      */
     @ConfigItem(defaultValue = "5S")
     public Duration connectionMaxIdleTime;
@@ -76,7 +84,8 @@ public class NettyHttpClientConfig {
     /**
      * Whether the idle connections in the connection pool should be closed.
      * <p>
-     * When enabled, connections left idling for longer than `quarkus.<amazon-service>.async-client.connection-max-idle-time`
+     * When enabled, connections left idling for longer than
+     * `quarkus.<amazon-service>.async-client.connection-max-idle-time`
      * will be closed. This will not close connections currently in use.
      */
     @ConfigItem(defaultValue = "true")
@@ -100,7 +109,7 @@ public class NettyHttpClientConfig {
      * Default is `OPENSSL` if available, `JDK` otherwise.
      */
     @ConfigItem
-    public Optional<SslProvider> sslProvider;
+    public Optional<SslProviderType> sslProvider;
 
     /**
      * HTTP/2 specific configuration
@@ -159,7 +168,8 @@ public class NettyHttpClientConfig {
         public OptionalInt initialWindowSize;
 
         /**
-         * Sets the period that the Netty client will send {@code PING} frames to the remote endpoint to check the
+         * Sets the period that the Netty client will send {@code PING} frames to the
+         * remote endpoint to check the
          * health of the connection. To disable this feature, set a duration of 0.
          * <p>
          * This setting is only respected when the HTTP/2 protocol is used.
@@ -181,16 +191,19 @@ public class NettyHttpClientConfig {
         /**
          * Number of threads to use for the event loop group.
          * <p>
-         * If not set, the default Netty thread count is used (which is double the number of available processors unless the
+         * If not set, the default Netty thread count is used (which is double the
+         * number of available processors unless the
          * `io.netty.eventLoopThreads` system property is set.
          */
         @ConfigItem
         public OptionalInt numberOfThreads;
 
         /**
-         * The thread name prefix for threads created by this thread factory used by event loop group.
+         * The thread name prefix for threads created by this thread factory used by
+         * event loop group.
          * <p>
-         * The prefix will be appended with a number unique to the thread factory and a number unique to the thread.
+         * The prefix will be appended with a number unique to the thread factory and a
+         * number unique to the thread.
          * <p>
          * If not specified it defaults to `aws-java-sdk-NettyEventLoop`
          */
@@ -210,14 +223,16 @@ public class NettyHttpClientConfig {
         /**
          * The endpoint of the proxy server that the SDK should connect through.
          * <p>
-         * Currently, the endpoint is limited to a host and port. Any other URI components will result in an exception being
+         * Currently, the endpoint is limited to a host and port. Any other URI
+         * components will result in an exception being
          * raised.
          */
         @ConfigItem
         public Optional<URI> endpoint;
 
         /**
-         * The hosts that the client is allowed to access without going through the proxy.
+         * The hosts that the client is allowed to access without going through the
+         * proxy.
          */
         @ConfigItem
         public Optional<List<String>> nonProxyHosts;
@@ -227,7 +242,8 @@ public class NettyHttpClientConfig {
     public static class Advanced {
 
         /**
-         * Whether the default thread pool should be used to complete the futures returned from the HTTP client request.
+         * Whether the default thread pool should be used to complete the futures
+         * returned from the HTTP client request.
          * <p>
          * When disabled, futures will be completed on the Netty event loop thread.
          */
@@ -235,6 +251,33 @@ public class NettyHttpClientConfig {
         public boolean useFutureCompletionThreadPool;
     }
 
-    //TODO: additionalChannelOptions
-    //additionalChannelOptions;
+    // TODO: additionalChannelOptions
+    // additionalChannelOptions;
+
+    /**
+     * An enumeration of SSL/TLS protocol providers.
+     */
+    public enum SslProviderType {
+
+        JDK {
+            @Override
+            SslProvider create() {
+                return io.netty.handler.ssl.SslProvider.JDK;
+            }
+        },
+        OPENSSL {
+            @Override
+            SslProvider create() {
+                return io.netty.handler.ssl.SslProvider.OPENSSL;
+            }
+        },
+        OPENSSL_REFCNT {
+            @Override
+            SslProvider create() {
+                return io.netty.handler.ssl.SslProvider.OPENSSL_REFCNT;
+            }
+        };
+
+        abstract SslProvider create();
+    }
 }

